src/browser-lib/cloudinary/extractCloudinaryIdFromURL.ts

```diff
@@ -0,0 +1,28 @@
+import OakError from "@/errors/OakError";
+
+/**
+ * Extracts the cloudinary id (public id and version if present)
+ * from a URL and returns it.
+ *
+ * `CldImage` and `OakCloudinaryImage` can automatically extract the Cloudinary id from a URL
+ * if it is under the target distribution origin. However currently the curriculum data has the full
+ * URL baked in which points to an origin under `cloudinary.com`. In order to construct a new URL under the
+ * new distribution inside `thenational.academy` we need to extract the Cloudinary id from the original URL
```

Comment:
☝🏼 not sure where this work belongs/when. Right now all other images inside OWA appear to be going through a vanilla Next `Image` so they are already being served from `thenational.academy` by virtue of them being proxied through Next.

Reply:
do you think it's okay to do the transformation in the image component? I don't like client-side data munging, but if the only ID source is the original URL I feel like it's that, or we add a field to the database, or we do it in the data layer (which would add complexity)

Reply:
🤔 I think i'm actually coming down on doing it in the data layer probably as a Zod transform (at least for now until we can decide what to do with the data). It would make this change more surgical and hide this detail from the client entirely.